lib/src/parser.dart

```diff
@@ -0,0 +1,222 @@
+
+    if (!map.containsKey('budgetUnit')) throw FormatException;
+    budget.budgetUnit = createBudgetUnit(map['budgetUnit']);
+
+    budget.automationType = createAutomationType(map['automationType']);
+
+    if (!map.containsKey('budgetSegments')) throw FormatException;
+    for (var map in map['budgetSegments']) {
+      budget.budgetSegments.add(createBudgetSegment(map));
+    }
+    return budget;
+  }
+
+  /// Creates an
+  /// [InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment]
+  /// instance from [map].
+  ///
+  /// Throws an [FormatException] if a required field is missing.
+  static InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment
+      createBudgetSegment(Map<String, dynamic> map) {
+    final budgetSegment =
+        InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment();
+
+    if (!map.containsKey('budgetAmountMicros')) throw FormatException;
+    budgetSegment.budgetAmountMicros = map['budgetAmountMicros'];
+
+    if (map.containsKey('description')) {
+      budgetSegment.description = map['description'];
+    }
+
+    if (!map.containsKey('dateRange')) throw FormatException;
+    budgetSegment.dateRange = createDateRange(map['dateRange']);
+
+    if (map.containsKey('campaignBudgetId')) {
+      budgetSegment.campaignBudgetId = map['campaignBudgetId'];
+    }
+    return budgetSegment;
+  }
+
+  /// Creates an
+  /// [InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange]
+  /// instance from [map].
+  static InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange
+      createDateRange(Map<String, dynamic> map) {
+    final dateRange =
+        InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange();
+
+    if (map.containsKey('startDate')) {
+      dateRange.startDate = createDate(map['startDate']);
+    }
+
+    if (map.containsKey('endDate')) {
+      dateRange.endDate = createDate(map['endDate']);
+    }
+
+    return dateRange;
+  }
+
+  /// Creates an
+  /// [InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange_Date]
+  /// from [map].
+  static InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange_Date
+      createDate(Map<String, dynamic> map) {
+    final date =
+        InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange_Date();
+    if (map.containsKey('year')) date.year = map['year'];
+    if (map.containsKey('month')) date.month = map['month'];
+    if (map.containsKey('day')) date.day = map['day'];
+    return date;
+  }
+
+  /// Creates an [InsertionOrder_EntityStatus] enum from [map].
+  ///
+  /// Throws an [FormatException] if [target] doesn't correspond
+  /// to any enum values.
+  static InsertionOrder_EntityStatus createEntityStatus(String target) {
+    for (var status in InsertionOrder_EntityStatus.values) {
+      if (status.name == target) {
+        return status;
+      }
+    }
+    throw FormatException();
```

Comment:
I think a better value is to return the 'unknown' or 'unspecified' default value, thats what our proto parses do in this case. This can happen as enums evolve over time and there is a mismatch between the client library and the server library.